src/BlockNode/BlockNode.ts

```diff
@@ -0,0 +1,57 @@
+import { EditorDocument } from '../EditorDocument';
+import { FormattingNode } from '../FormattingNode';
+import { TextNode } from '../TextNode';
+import { ValueNode } from '../DataNode';
+import { BlockTune } from '../BlockTune';
+import { BlockNodeConstructorParameters } from './types';
+
+/**
+ * BlockNode class represents a node in a tree-like structure used to store and manipulate content in an editor document.
+ * A BlockNode can contain one or more child nodes of type TextNode, DataNode or FormattingNode.
+ * It can also be associated with one or more BlockTunes, which can modify the behavior of the BlockNode.
+ */
+export class BlockNode {
+  /**
+   * Private field representing the name of the BlockNode
+   *
+   * @private
+   */
+  #name: string;
```

Comment:
why not `private` ?

Reply:
It's a private field.

See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields

![](https://www.typescriptlang.org/static/TypeScript%20Classes-83cc6f8e42ba2002d5e2c04221fa78f9.png)

Reply:
Why it is better than TypeScript notation?

Reply:
Because after compilation your private field isn't private:
<img width="1402" alt="image" src="https://user-images.githubusercontent.com/37909603/230959547-9a62d492-d509-45e1-8149-867893ce5a1a.png">

And if you use #, TS creates polyfill:
<img width="1397" alt="image" src="https://user-images.githubusercontent.com/37909603/230959676-ab1046d2-fde8-47a4-85a3-1d16f44c4925.png">
or uses native JS (for ES2022+):
<img width="373" alt="image" src="https://user-images.githubusercontent.com/37909603/230959813-1aced7ef-e269-4e05-b07d-de2741a25e2a.png">